### PR TITLE
attribute checkers

### PR DESCRIPTION
**Pull request by Dragos0000**

### CODE REVIEW

File: src/html-conventions-lib/class-attributes-with-same-name.xsl [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            name</xd:desc>
+        <xd:param name="attributeName"/>
+        <xd:param name="root"/>
+    </xd:doc>
+    <xsl:template name="checkDefinitionOfAttributesWithSameName">
+        <xsl:param name="attributeName"/>
+        <xsl:param name="root"/>
+        <xsl:variable name="attributesWithSameName"
+            select="f:getClassAttributeByName($attributeName, $root)"/>
+        <xsl:variable name="definitionValues" select="$attributesWithSameName/documentation/@value"/>
```
Comment (costezki) [inline src/html-conventions-lib/class-attributes-with-same-name.xsl:93]: do we need `definitionValues` and `descriptionsWithAnnotations` and the next ?

File: src/html-conventions-lib/class-attributes-with-same-name.xsl [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    if ($attribute/documentation/@value) then
+                        fn:concat($attribute/documentation/@value, ' (', $attribute/../../@name, ') ')
+                    else
+                        ()"/>
+
+        <xsl:variable name="allAttributesHaveDefinition"
+            select="fn:count($attributesWithSameName) = fn:count($definitionValues)"/>
+        <xsl:sequence
+            select="
+                if (f:areStringsEqual($definitionValues) and fn:boolean($definitionValues) and $allAttributesHaveDefinition) then
```
Comment (costezki) [inline src/html-conventions-lib/class-attributes-with-same-name.xsl]: :) Funky logics :)

File: src/html-conventions-lib/class-attributes-with-same-name.xsl [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                    f:generateHtmlInfo(fn:concat('The property is reused in multiple contexts, the meaning given by the definition is the same.  ',
```
Comment (costezki) [inline src/html-conventions-lib/class-attributes-with-same-name.xsl]: do we need this info? I would say we don't. This is the norm.

File: src/html-conventions-lib/class-attributes-with-same-name.xsl (hunk 2 above)
Comment (costezki) [reply src/html-conventions-lib/class-attributes-with-same-name.xsl]: most likely can be simplified.

**Review by costezki — APPROVED** [review]
I wonder if the checker logic can be simplified.

**Review by costezki — CHANGES_REQUESTED** [review]
Just checked the warning messages of the checkers, and they are hard to read. 

I would recommend that we Upgrade the functions/templates that generate the warning/info messages so that they support HTML formatting. This will allow us to build the more readable text. 

Next, the attribute and connector definitions must be pre-processed using the `formatDocumentation` function which will transform EA specific tags into HTML tags making them more readable. 

In addition, the list of attribute types  (in the warning for attributes with multiple data type usage) should be a vertical bullet list for increased readability.

**Comment by costezki** [comment]
I just checked the `f:generateHtmlError` and `f:generateHtmlWarning` and they seem to be ready for HTML formatted messages. If not then this would be a lead to fix the issue: 
I noticed output is generated using the `xsl:value-of` which might be forcing any content to be cast into pure text. The alternative would be to switch to `xsl:sequance` instead. Optionally we could employ a <div> content wrapper just in case we need a mechanism to hold the content together. The later, however, might break some of the current tests.

**Review by costezki — APPROVED** [review]
fine, let's move the task into a new branch

**Review by costezki — APPROVED** [review]
fine, let's move the task into a new branch

File: src/html-conventions-lib/class-attributes-with-same-name.xsl (hunk 2 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-attributes-with-same-name.xsl]: took of the info

File: src/html-conventions-lib/class-attributes-with-same-name.xsl (hunk 3 above)
Comment (Dragos0000) [reply src/html-conventions-lib/class-attributes-with-same-name.xsl]: nope

**Pull request merged**